Dockerfile

```diff
@@ -0,0 +1,24 @@
+# Miden-node Dockerfile
+
+#### Setup image builder
+FROM rust:1.75.0-bullseye AS builder
+
+# Install dependencies
+RUN apt-get update && apt-get -y upgrade
+RUN apt-get install -y gcc llvm clang bindgen pkg-config
+
+# Setup workdir
+WORKDIR /app
+COPY . miden-node
+RUN cd miden-node && make
+
+### Run Miden-Node
+FROM ubuntu:22.04
```

Comment:
questions:

- why ubuntu here instead of debian like the builder image? if we can, debian slim would be better
- why 22.04 instead of 24.04?

Reply:
This image should have [`LABEL`s](https://docs.docker.com/reference/dockerfile/#label). I suggest using [open containers labels](https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys):

```
LABEL org.opencontainers.image.authors=miden@polygon.io \
      org.opencontainers.image.url=https://0xpolygonmiden.github.io/ \
      org.opencontainers.image.documentation=https://github.com/0xPolygonMiden/miden-node \
      org.opencontainers.image.source=https://github.com/0xPolygonMiden/miden-node \
      org.opencontainers.image.vendor=Polygon \
      org.opencontainers.image.licenses=MIT

ARG CREATED
ARG VERSION
ARG COMMIT
LABEL org.opencontainers.image.created=$CREATED \
      org.opencontainers.image.version=$VERSION \
      org.opencontainers.image.revision=$COMMIT
```

Reply:
We need to add a [`VOLUME`](https://docs.docker.com/reference/dockerfile/#volume) to have the store's db.

Reply:
To add a volume we would need to modify where the node reads and writes it's database files in order to persist them to the volume.

Reply:
> To add a volume we would need to modify where the node reads and writes it's database files in order to persist them to the volume.


The [`WORKDIR`](https://docs.docker.com/reference/dockerfile/#workdir) setting should handle that.

Reply:
This is the last thing pending, how not yet been able to make it work correctly. Looking into it now.

Reply:
Managed to fix this, now the node creates a db folder to put it's sqlite files, this volume is mounted from the host as a volume.